packages/angular/projects/ui-angular/src/lib/components/authenticator/components/confirm-verify-user/amplify-confirm-verify-user.component.html

```diff
@@ -6,7 +6,7 @@
       data-amplify-fieldset
       [disabled]="authenticator.isPending"
     >
-      <amplify-slot name="confirm-verify-user-header">
+      <amplify-slot name="confirm-verify-user-header" [context]="context">
```

Comment:
How does Angular make it so deeply nested slots like this, is available to the `authenticator` component? In vue this slot would only be available in the confirm verify user component.

Reply:
Good question, I used `CustomComponentsService` to transfer those slots into deeply nested slots.


1. All `amplifySlot` templates passed into top-level `<amplify-authenticator` are provided to `CustomComponentsService`

https://github.com/aws-amplify/amplify-ui/blob/ca1b0b8bb686eaf61e5ba5899b5f3e2e19c73ad9/packages/angular/projects/ui-angular/src/lib/components/authenticator/components/authenticator/authenticator.component.ts#L74-L76

2. `amplify-slot` component, wherever it's placed, uses that service to fetch the `ng-template` and renders it:

https://github.com/aws-amplify/amplify-ui/blob/ca1b0b8bb686eaf61e5ba5899b5f3e2e19c73ad9/packages/angular/projects/ui-angular/src/lib/utilities/amplify-slot/amplify-slot.component.ts#L25-L33

so this is agnostic of where it's placed.

Reply:
👍